pandas/tests/frame/methods/test_reset_index.py

```diff
@@ -379,6 +379,14 @@ def test_reset_index_multiindex_columns(self):
         with pytest.raises(ValueError, match=msg):
             df.rename_axis("A").reset_index()
 
+        # GH#44755 reset_index with duplicate column labels
```

Comment:
can you make a separate test, also test with option_context and setting the option

Reply:
I have made a separate test, and included tests of the DataFrame flag (which was tricky, because many things stop working if the flag is set to False! E.g. rename_axis())
It is a flag, not a global option, so I don't think that option_context is appropriate.